src/test/scala/uk/gov/ons/registers/stepdefs/SamplingSteps.scala

```diff
@@ -1,76 +1,61 @@
-    assertNewCellNumberFieldHasBeenAdded(prnSampleCsvFile)
-
-    val expectedPrnSampleFirstRow = expectedFirstRow(cellNumber="5813")
-    assertRowEquality(csvRow = enterpriseRowFromMap(csvRowAsColumnsWithMap = lineWithHeaderAsMap(prnSampleCsvFile)), expectedRow = expectedPrnSampleFirstRow)
+  When("""a Scala Sample is created from a Stratified Frame"""){ () =>
+    createSampleTest()
+    outputDataDF = outputDataDF.na.fill(value = "")
+  }
 
-    displayData(expectedRow = expectedPrnSampleFirstRow)
+  When("""an exception in Scala is thrown for Stratified Properties not being found upon trying to Sample"""){ () =>
+    assert(aFailureIsGeneratedBy {
+      createSampleTest()
+    })
   }
 
-  Then("""a Sample DataFrame is returned and exported to CSV with the inclusion of stratas with outbound Sample Size parameter"""){ () =>
-    val prnSampleWithOutOfBoundsSampleSizeCsvFile = assertAndReturnCsvOfSampleCollection
-    assertSampleCollectionSize(sampleCollectionCsv = prnSampleWithOutOfBoundsSampleSizeCsvFile, expectedNumberOfRecords = 2)
-    assertNewCellNumberFieldHasBeenAdded(prnSampleWithOutOfBoundsSampleSizeCsvFile)
+  When("""an exception in Scala is thrown for Stratified Frame not being found upon trying to Sample"""){ () =>
+    assert(aFailureIsGeneratedBy {
+      createSampleTest()
+    })
+  }
 
-    val firstRowPrn = outputDataDF.first.getAs[String](prn)
-    val secondRowPrn = outputDataDF.take(2).last.getAs[String](prn)
-    assert(BigDecimal(firstRowPrn) > BigDecimal(secondRowPrn), message = s"expected prn of first row [$firstRowPrn] in DataFrame to be higher than prn of second row [$secondRowPrn]")
+  Then("""a Sample containing the sample selection from the Census strata is returned and exported to CSV"""){ theExpectedResult: DataTable =>
+    assertDataFrameEquality(expected = theExpectedResult, printLabel)
   }
 
-  Then("""a Sample DataFrame is returned and exported to CSV with the invalid Sample Size strata logged and entire Frame returned for that strata"""){ () =>
-    val sampleSizeTooGreatCsvFile = assertAndReturnCsvOfSampleCollection
-    assertSampleCollectionSize(sampleCollectionCsv = sampleSizeTooGreatCsvFile, expectedNumberOfRecords = frameSize)
-    assertNewCellNumberFieldHasBeenAdded(sampleSizeTooGreatCsvFile)
+  Then("""a Sample containing the Sample Size from the Prn-Sample strata is returned and exported to CSV"""){ theExpectedResult: DataTable =>
+    assertDataFrameEquality(expected = theExpectedResult, printLabel)
+  }
 
-    val expectedSampleWithSampleSizeTooGreatFirstRow = expectedFirstRow(cellNumber="5899")
-    assertRowEquality(csvRow = enterpriseRowFromMap(csvRowAsColumnsWithMap = lineWithHeaderAsMap(sampleSizeTooGreatCsvFile)),
-      expectedRow = expectedSampleWithSampleSizeTooGreatFirstRow)
+  Then("""a Sample containing the Sample Size from the PRN-sampling strata is returned and exported to CSV"""){ theExpectedResult: DataTable =>
```

Comment:
@pwetherill I think this 'Then' could be tweaked to convey that it is looping the dataframe sequentially and thereby making it a bit little different than the above Then